addons/pos_loyalty/static/src/overrides/models/pos_store.js

```diff
@@ -298,18 +298,12 @@ patch(PosStore.prototype, {
                 return;
             }
             order.invalidCoupons = false;
-            const allCoupons = [];
-            for (const pe of Object.values(order.uiState.couponPointChanges)) {
-                if (pe.coupon_id > 0) {
-                    allCoupons.push(pe.coupon_id);
-                }
-            }
-            allCoupons.push(...order._code_activated_coupon_ids.map((coupon) => coupon.id));
-            const couponsToFetch = allCoupons.filter(
-                (elem) => !this.models["loyalty.card"].get(elem)
-            );
+
+            const couponsToFetch = this.models["loyalty.card"]
+                .filter((card) => card.partner_id)
+                .map((card) => card.id);
```

Comment:
Maybe I'm missing something, but the aim here is to retrieve missingCoupon, and then you look at the existing coupons to retrieve it again if there's a partner ID?